cfd_protocol/src/protocol.rs

```diff
@@ -369,6 +398,7 @@ impl Payout {
                 digits,
                 maker_amount,
                 taker_amount,
+                event_id: event_id.to_string(),
```

Comment:
Nit: It is not ideal to accept a `&str` if you are going to unconditionally call `to_string` on it because it means you are forcing an allocation even if the caller would already have an owned instance they could give you ownership of.

If you want flexibility in passing `&str` and `String`, use `impl Into<String>` as a type, that will evaluate to a no-op for `String`.

Reply:
I thought we need to allocate anyway because we have an iterator over digits.

Reply:
Perhaps but that iterator has nothing to do with this string. Allocations are not batched together when they concern different variables.